staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -158,25 +175,58 @@ func (q *delayingType) ShutDown() {
 	})
 }
 
+// Add sends an item for immediate queuing, preempting a waiting item as it is queued.
+func (q *delayingType) Add(item interface{}) {
+	q.addToDelayedQueue(item, 0, DelayedAddOptions{})
```

Comment:
if this is going to preempt a waiting item, shouldn't it set `DropIfWaiting: true`? since it is not setting this currently, is addToDelayedQueue doing the right thing?

Reply:
oh, maybe I misunderstood what `DropIfWaiting` does... it's ambiguous whether the thing that is dropped is the waiting item or the added item

Reply:
Ah yes, have made an effort to improve the godoc of the functions and types